### PR TITLE
add new case: uefi_check_debugcon

### PR DESCRIPTION
**Pull request by XueqiangWei**
Verify the "-debugcon" parameter under the UEFI environment.

ID: 1813922
Signed-off-by: Xueqiang Wei <xuwei@redhat.com>

### CODE REVIEW

**Comment by XueqiangWei** [comment]
(1/2) Host_RHEL.m8.u2.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.uefi_check_debugcon.disable.q35: PASS (112.11 s)
 (2/2) Host_RHEL.m8.u2.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.uefi_check_debugcon.enable.q35: PASS (104.75 s)

**Comment by XueqiangWei** [comment]
@zhencliu please help review it. Many thanks.

File: qemu/tests/uefi_check_debugcon.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            return True
+
+    def remove_isa_debugcon(vm):
+        """
+        remove the existing isa-log device
+        """
+        for device in vm.devices:
+            if device.type == "isa-log":
+                vm.devices.remove(device)
+                env.register_vm(vm.name, vm)
```
Comment (xutian) [inline qemu/tests/uefi_check_debugcon.py]: Hi, no need to register VM in loop, this line should be in out-side of for loop

File: qemu/tests/uefi_check_debugcon.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    def trace_kvm_pio():
+        """
+        trace event kvm_pio
+        """
+        process.system(trace_record_cmd)
+
+    error_context.context("Check whether trace-cmd has been "
+                          "installed on host.", logging.info)
+    if process.system("which trace-cmd", ignore_status=True):
+        test.cancel("Please install trace-cmd on host.")
```
Comment (xutian) [inline qemu/tests/uefi_check_debugcon.py]: test.error instead test.cancel

File: qemu/tests/uefi_check_debugcon.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        vm.verify_alive()
+        vm.destroy()
+        process.system(stop_trace_record, ignore_status=True, shell=True)
+        if not utils_misc.wait_for(
+                lambda: not check_trace_process(), timeout, 30, 3):
+            test.fail("Failed to stop command: '%s' after %s seconds."
+                      % (stop_trace_record, timeout))
+        pio_read_counts = process.run(
+                check_pio_read, shell=True).stdout.decode().strip()
+        if not eval(pio_read_counts + params["pio_read_counts"]):
```
Comment (xutian) [inline qemu/tests/uefi_check_debugcon.py]: don't do this, please count  pio_read_counts in new line.

File: qemu/tests/uefi_check_debugcon.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            test.fail("pio_read counts should be greater than 0. "
+                      "But the actual counts are %s." % pio_read_counts)
+        pio_write_counts = process.run(
+                check_pio_write, shell=True).stdout.decode().strip()
+        if not eval(pio_write_counts + params["pio_write_counts"]):
```
Comment (xutian) [inline qemu/tests/uefi_check_debugcon.py]: don't use eval here.

File: qemu/tests/uefi_check_debugcon.py (hunk 3 above)
Comment (xutian) [reply qemu/tests/uefi_check_debugcon.py]: test.assertLessEqual(pio_read_count, 0)

File: qemu/tests/uefi_check_debugcon.py (hunk 4 above)
Comment (xutian) [reply qemu/tests/uefi_check_debugcon.py]: test.assertEqual(pio_read_count, 0)

File: qemu/tests/uefi_check_debugcon.py (hunk 1 above)
Comment (XueqiangWei) [reply qemu/tests/uefi_check_debugcon.py]: updated, thanks.

File: qemu/tests/uefi_check_debugcon.py (hunk 2 above)
Comment (XueqiangWei) [reply qemu/tests/uefi_check_debugcon.py]: updated, thanks

File: qemu/tests/uefi_check_debugcon.py (hunk 3 above)
Comment (XueqiangWei) [reply qemu/tests/uefi_check_debugcon.py]: updated, thanks.

File: qemu/tests/uefi_check_debugcon.py (hunk 4 above)
Comment (XueqiangWei) [reply qemu/tests/uefi_check_debugcon.py]: updated, thanks.

**Comment by XueqiangWei** [comment]
(1/2) Host_RHEL.m8.u2.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.uefi_check_debugcon.disable.q35: PASS (112.31 s)
 (2/2) Host_RHEL.m8.u2.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.uefi_check_debugcon.enable.q35: PASS (106.39 s)

@zhencliu @xutian please help review it again. Many thanks.

**Comment by xutian** [comment]
ACK

File: qemu/tests/uefi_check_debugcon.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+            check_pio_write, shell=True).stdout.decode().strip())
+        if params.get("ovmf_log"):
+            err_str = "pio_write counts should be greater than 0. "
+            err_str += "But the actual counts are %s." % pio_write_counts
+            test.assertGreater(pio_write_counts, 0, err_str)
+        else:
+            err_str = "pio_write counts should be equal to 0. "
+            err_str += "But the actual counts are %s." % pio_write_counts
+            test.assertEqual(pio_write_counts, 0, err_str)
+    finally:
```
Comment (zhencliu) [inline qemu/tests/uefi_check_debugcon.py]: remove ovmf_log and trace_output_file if they exist

File: qemu/tests/uefi_check_debugcon.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+    def trace_kvm_pio():
+        """
+        trace event kvm_pio
+        """
+        process.system(trace_record_cmd)
+
+    error_context.context("Check whether trace-cmd has been "
+                          "installed on host.", logging.info)
+    if process.system("which trace-cmd", ignore_status=True):
```
Comment (zhencliu) [inline qemu/tests/uefi_check_debugcon.py]: cmds_installed_host += 'trace-cmd', so the check here can be removed, just fyi.

File: qemu/tests/uefi_check_debugcon.py (hunk 5 above)
Comment (XueqiangWei) [reply qemu/tests/uefi_check_debugcon.py]: Don't remove them, put them into test.debugdir. It's a part of logs.  thanks

File: qemu/tests/uefi_check_debugcon.py (hunk 6 above)
Comment (XueqiangWei) [reply qemu/tests/uefi_check_debugcon.py]: I think all work well. thank you.

**Comment by zhencliu** [comment]
LGTM

**Pull request merged**